src/app/utils/route-utils.ts

```diff
@@ -130,8 +141,13 @@ export abstract class RouteUtils {
     return `${this.getScopeRoute(scope)}/block`
   }
 
-  static getBlockRoute = (scope: SearchScope, blockHeight: number) => {
-    return `${this.getScopeRoute(scope)}/block/${encodeURIComponent(blockHeight)}`
+  static getBlockRoute = (
+    scope: SearchScope,
+    blockHeight: number,
+    searchParams?: URLSearchParams,
+    preserveParams: string[] = [],
```

Comment:
I hate it

Reply:
Would you prefer a more generally useful `options` object, for which 'preserveParams` would be just one possilble key?

Reply:
no. Ideally I wouldn't preserve params. But if we want to, then getBlockRoute should be the function that decides which params to preserve